### ISSUE TITLE
SharePlusPlugin.java uses or overrides a deprecated API.

### ISSUE THREAD

**Issue by sucasuc** [fluttercommunity/plus_plugins#314]
## System info

Issue occurs on: Android
Plugin name: share_plus
Plugin version: 2.1.4
Flutter doctor output:

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-x64, locale en-AD)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] VS Code (version 1.57.1)
[✓] Connected device (3 available)
• No issues found!

Step to reproduce

When i build android app using command, get a warning
```
flutter build appbundle
Note: /Users/admin/Development/flutter/.pub-cache/hosted/pub.dartlang.org/share_plus-2.1.4/android/src/main/java/dev/fluttercommunity/plus/share/SharePlusPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.          
```

**Comment by mhadaily** [comment]
possible duplication #335

**Comment by Coronon** [comment]
Running with `-Xlint:deprication` reveals the following:
```
...\Flutter\.pub-cache\hosted\pub.dartlang.org\share_plus-2.1.4\android\src\main\java\dev\fluttercommunity\plus\share\SharePlusPlugin.java:14: warning: [deprecation] Registrar in PluginRegistry has been deprecated
import io.flutter.plugin.common.PluginRegistry.Registrar;
                                              ^
...\plus\share\SharePlusPlugin.java:24: warning: [deprecation] Registrar in PluginRegistry has been deprecated
  public static void registerWith(Registrar registrar) {
```

This is caused by share_plus still using V1 embedding, which is [depreacted](https://api.flutter.dev/javadoc/io/flutter/plugin/common/PluginRegistry.Registrar.html). Docs on embedding V2 and migration can be found [here](https://flutter.dev/docs/development/packages-and-plugins/plugin-api-migration).

**Comment by sashkent3** [comment]
There seems to be this issue with package_info_plus and device_info_plus as well.

**Comment by mhadaily** [comment]
@Coronon please go ahead and send us a PR. sorry for my delay to answer. 

btw, we also have released a few new version, please check them out.

**Comment by Coronon** [comment]
@mhadaily I’ll work on this once I have time on my hands (probably not before the weekend)

**Comment by labaxter** [comment]
This is still an issue.

**Comment by mhadaily** [comment]
@Coronon how far you are on this?

**Comment by Coronon** [comment]
@mhadaily @labaxter Not very far I am afraid. There were a few more pressing issues that I worked on: #765, #800, #831 and now #847 has not been reviewed in 20 days :/ I would like that to be merged before starting to work on any new PR.

**Comment by mhadaily** [comment]
I am sorry for the delay to review all of us here have been very very busy with Google I/O, we had some deliverables for the Google I/O. @Coronon 

Don't let that stop you to continue your amazing work. we appreciate it. 
Next week, I will answer so many on-hold messages and PRs that I have on my list, including sending swag to continuators in April.

**Comment by Coronon** [comment]
After some investigation I can not reproduce this issue, it was likely fixed by #495 (which you were involved into @mhadaily) but at latest with #666.

If no one else can reproduce this -> Please close :)

**Comment by mhadaily** [comment]
@labaxter can you please confirm?
